content-services/content-sources-backend
# Fixes 4293: add support for upload only repos

## Summary

## Testing steps

Create an upload repo:

```
PATCH http://localhost:8000/api/content-sources/v1.0/repositories/c06a0512-0954-4b11-9b13-3358afe77f46
x-rh-identity: eyJpZGVudGl0eSI6eyJvcmdfaWQiOiI5IiwgInR5cGUiOiJVc2VyIiwidXNlciI6eyJ1c2VybmFtZSI6ImZvbyJ9LCJhY2NvdW50X251bWJlciI6ImZvbyIsImludGVybmFsIjp7Im9yZ19pZCI6IjkifX19Cg==
//x-Rh-Insights-Request-Id: 9d229d34-7219-405d-ba3d-8f99cf7c36ae
Content-Type: application/json

{
  "name": "arg",
  "origin": "upload"
}
```

with batch:
```
POST http://localhost:8000/api/content-sources/v1.0/repositories/bulk_create/
x-rh-identity: eyJpZGVudGl0eSI6eyJvcmdfaWQiOiI5IiwgInR5cGUiOiJVc2VyIiwidXNlciI6eyJ1c2VybmFtZSI6ImZvbyJ9LCJhY2NvdW50X251bWJlciI6ImZvbyIsImludGVybmFsIjp7Im9yZ19pZCI6IjkifX19Cg==

[{
  "name": "test5",
  "gpg_key": "",
  "snapshot": true,
  "origin": "upload",
  "url": ""
},
  {
    "name": "test6",
    "gpg_key": "",
    "snapshot": true,
    "origin": "upload",
    "url": ""
  }]

```

try to:

1.  create/update an upload repo with a url, this should fail
2. try to change the origin of a repo, this should not succeed
3. try to introspect an upload repo, this should be rejected
4.  try to snapshot an upload repo, this should be rejected 
5.  run `go run cmd/external-repos/main.go nightly-jobs` and verify no errors, that things work as expected

## Checklist
